### PR TITLE
[feat/fix/docs]: Improvements in the `backtracking` folder

### PR DESCRIPTION
**Pull request by Panquesito7**
#### Checklist

- [x] Added description of the change
- [ ] ~Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)~
- [ ] ~Added tests and example, the test must pass~
- [ ] ~Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)~
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: ~I'll try to fix the merge conflicts in this PR. The reason for this is because I previously worked on this, but I didn't submit the PR the same day I started working on it~ The merge conflicts are resolved now.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1553"></a>

### CODE REVIEW

File: backtracking/graph_coloring.cpp [hunk 1]
```diff
@@ -17,29 +17,38 @@
...
  */
 template <size_t V>
 void printSolution(const std::array<int, V>& color) {
-    std::cout << "Following are the assigned colors" << std::endl;
+    std::cout << "Following are the assigned colors\n";
     for (auto& col : color) {
         std::cout << col;
     }
-    std::cout << std::endl;
+    std::cout << "\n";
```
Comment (mishraabhinn) [inline backtracking/graph_coloring.cpp:47]: Why?

**Review by mishraabhinn — CHANGES_REQUESTED** [review]

File: backtracking/graph_coloring.cpp (hunk 1 above)
Comment (Panquesito7) [reply backtracking/graph_coloring.cpp:47]: According to what @ayaankhan98 has said previously, `\n` is faster than `std::endl`.

**Review by ayaankhan98 — APPROVED** [review]
Try not to improve more than 2 files per PR. Keep the PR as much concise as possible.

**Review by mishraabhinn — APPROVED** [review]

**Comment by saurabhprasadsah** [comment]
yes

**Pull request merged**